Sources/Amplitude/Amplitude.swift

```diff
@@ -299,8 +308,8 @@ public class Amplitude {
 
     @discardableResult
     public func reset() -> Amplitude {
-        _ = setUserId(userId: nil)
-        _ = setDeviceId(deviceId: nil)
+        setUserId(userId: nil)
+        setDeviceId(deviceId: nil)
```

Comment:
Is it possible that the contextPlugin.initializeDeviceId is called before setDeviceId got a chance to execute resulting in a null device id?

Reply:
Both userId/deviceId calls are sync, but there is a slight window in which two calls to reset() from different threads could have a nil deviceId set in storage. I've removed the call to `setDeviceId(deviceId: nil)` and added a forceReset parameter to the existing `contextPlugin.initializeDeviceId` so it is only set once.